image/serde/src/lib.rs

```diff
@@ -28,7 +29,40 @@ impl<W: Write> ImageBundleWriter<W> {
 
     /// Write Image Bundle
     pub fn write(&mut self, image: &ImageBundle) -> anyhow::Result<()> {
-        self.writer.write_all(image.manifest.as_bytes())?;
+        // Add fields till the ECC vendor key descriptor.
+        let offset = offset_of!(ImageManifest, preamble);
+        let ptr = &image.manifest as *const _ as *const u8;
+        let slice = unsafe { std::slice::from_raw_parts(ptr, offset) };
```

Comment:
This feels unnecessarily unsafe. Why not `self.writer.write_all(image.manifest.preamble.as_bytes())`?

Reply:
That cannot be done since Preamble has static sized ECC and LMS key arrays.